packages/react-internal/src/components/TeachingBubble/TeachingBubble.base.tsx

```diff
@@ -1,103 +1,88 @@
+import { classNamesFunction } from '../../Utilities';
 import { TeachingBubbleContent } from './TeachingBubbleContent';
 import {
+  ITeachingBubble,
   ITeachingBubbleProps,
   ITeachingBubbleStyleProps,
   ITeachingBubbleStyles,
   ITeachingBubbleSubComponentStyles,
 } from './TeachingBubble.types';
 import { Callout, ICalloutProps } from '../../Callout';
 import { DirectionalHint } from '../../common/DirectionalHint';
+import { useMergedRefs } from '@uifabric/react-hooks';
 
-const getClassNames = classNamesFunction<ITeachingBubbleStyleProps, ITeachingBubbleStyles>();
+const COMPONENT_NAME = 'TeachingBubble';
+const defaultCalloutProps: ICalloutProps = {
+  beakWidth: 16,
+  gapSpace: 0,
+  setInitialFocus: true,
+  doNotLayer: false,
+  directionalHint: DirectionalHint.rightCenter,
+};
 
-export interface ITeachingBubbleState {
-  isTeachingBubbleVisible?: boolean;
-}
+const getClassNames = classNamesFunction<ITeachingBubbleStyleProps, ITeachingBubbleStyles>();
+const useComponentRef = (
+  componentRef: React.Ref<ITeachingBubble> | undefined,
+  rootElementRef: React.RefObject<HTMLDivElement>,
+) => {
+  React.useImperativeHandle(
+    componentRef,
+    () => ({
+      focus: () => rootElementRef.current?.focus(),
+    }),
+    [rootElementRef],
+  );
+};
 
-export class TeachingBubbleBase extends React.Component<ITeachingBubbleProps, ITeachingBubbleState> {
-  public static defaultProps = {
-    /**
-     * Default calloutProps is deprecated in favor of private `_defaultCalloutProps`.
-     * Remove in next release.
-     * @deprecated In favor of private `_defaultCalloutProps`.
-     */
+export const TeachingBubbleBase: React.FunctionComponent<ITeachingBubbleProps> = React.forwardRef<
+  HTMLDivElement,
+  ITeachingBubbleProps
+>((props, forwardedRef) => {
+  const rootElementRef = React.useRef<HTMLDivElement>(null);
+  const mergedRootRef = useMergedRefs(rootElementRef, forwardedRef);
+  const {
+    calloutProps: setCalloutProps,
     // eslint-disable-next-line deprecation/deprecation
-    calloutProps: {
-      beakWidth: 16,
-      gapSpace: 0,
-      setInitialFocus: true,
-      doNotLayer: false,
-      directionalHint: DirectionalHint.rightCenter,
-    },
-  };
-
-  public rootElement = React.createRef<HTMLDivElement>();
-  private _defaultCalloutProps: ICalloutProps;
-
-  // Constructor
-  constructor(props: ITeachingBubbleProps) {
-    super(props);
-
-    initializeComponentRef(this);
-    this.state = {};
+    targetElement,
+    onDismiss,
+    // eslint-disable-next-line deprecation/deprecation
+    hasCloseButton = props.hasCloseIcon,
+    isWide,
+    styles,
+    theme,
+    target,
+  } = props;
 
-    this._defaultCalloutProps = {
-      beakWidth: 16,
-      gapSpace: 0,
-      setInitialFocus: true,
-      doNotLayer: false,
-      directionalHint: DirectionalHint.rightCenter,
-    };
-  }
+  const calloutProps = { ...defaultCalloutProps, ...setCalloutProps };
```

Comment:
Just realized, this needs to be memoized to avoid constant style re-calculation. You can do it like this to also preserve decent type safety:
```ts
  const calloutProps: ICalloutProps = React.useMemo(() => ({ ...defaultCalloutProps, ...setCalloutProps }), [
    setCalloutProps,
  ]);

  const stylesProps: ITeachingBubbleStyleProps = {
    theme: theme!,
    isWide,
    calloutProps: calloutProps as ICalloutProps & Required<Pick<ICalloutProps, 'theme'>>,
    hasCloseButton,
  };
```

Reply:
Realized my initial suggestion was slightly wrong, you do have to mix in theme manually. I pushed a fix.